app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/backup/RestoreFromBackupViewModel.kt

```diff
@@ -184,39 +176,129 @@ class RestoreFromBackupViewModel @Inject constructor(
     }
 
     fun onContinueClick() = viewModelScope.launch {
-        if (state.value.isRestoringProfileChecked) {
-            sendEvent(Event.OnRestoreConfirm(fromCloud = true))
+        val profileToRestore = state.value.restoringProfiles?.first { selectable -> selectable.selected }
+
+        profileToRestore?.let { selectableRestoringProfile ->
+            if (selectableRestoringProfile.data.isFromGoogleDrive && selectableRestoringProfile.data.googleDriveFileId != null) {
+                _state.update { it.copy(isDownloadingSelectedCloudBackup = true) }
+
+                val entity = CloudBackupFileEntity(
+                    id = selectableRestoringProfile.data.googleDriveFileId,
+                    profileId = selectableRestoringProfile.data.profileId,
+                    lastUsedOnDeviceName = selectableRestoringProfile.data.deviceDescription,
+                    lastUsedOnDeviceModified = selectableRestoringProfile.data.lastModified,
+                    totalNumberOfAccountsOnAllNetworks = selectableRestoringProfile.data.totalNumberOfAccountsOnAllNetworks,
+                    totalNumberOfPersonasOnAllNetworks = selectableRestoringProfile.data.totalNumberOfPersonasOnAllNetworks
+                )
+
+                downloadBackedUpProfileFromCloud(
+                    entity = entity
+                ).fold(
+                    onSuccess = { cloudBackupFile ->
+                        val backupType = BackupType.Cloud(entity)
+                        saveTemporaryRestoringSnapshotUseCase.forCloud(cloudBackupFile.serializedProfile, backupType)
+                        _state.update { it.copy(isDownloadingSelectedCloudBackup = false) }
+                        sendEvent(Event.OnRestoreConfirmed(backupType))
+                    },
+                    onFailure = { exception ->
+                        _state.update {
+                            it.copy(
+                                uiMessage = UiMessage.ErrorMessage(exception),
+                                isDownloadingSelectedCloudBackup = false
+                            )
+                        }
+                    }
+                )
+            } else {
+                sendEvent(Event.OnRestoreConfirmed(BackupType.DeprecatedCloud))
+            }
         }
     }
 
     fun onMessageShown() = _state.update { it.copy(uiMessage = null) }
 
-    private suspend fun restoreProfileFromCloudBackup() {
-        getTemporaryRestoringProfileForBackupUseCase(BackupType.Cloud)?.let { restoringProfile ->
-            if (restoringProfile.header.isCompatible) {
-                _state.update {
-                    it.copy(restoringProfiles = listOf(restoringProfile))
-                }
+    private suspend fun restoreProfilesFromCloudBackup() {
+        val availableCloudBackedUpProfiles = fetchBackedUpProfilesMetadataFromCloud()
+            .onFailure {
```

Comment:
Maybe we should log this to crashlytics as non-fatal, if that would have any value in potential backup troubleshooting?